cni/plugin.go

```diff
@@ -198,3 +199,43 @@ func (plugin *Plugin) UninitializeKeyValueStore() error {
 
 	return nil
 }
+
+// check if safe to remove lockfile
+func (plugin *Plugin) IsSafeToRemoveLock(processName string) (bool, error) {
+	if plugin != nil && plugin.Store != nil {
+		// check if get process command supported
+		if cmdErr := platform.GetProcessSupport(); cmdErr != nil {
+			log.Errorf("Get process cmd not supported. Error %v", cmdErr)
+			return false, cmdErr
+		}
+
+		// Read pid from lockfile
+		lockFileName := plugin.Store.GetLockFileName()
+		content, err := ioutil.ReadFile(lockFileName)
+		if err != nil {
+			log.Errorf("Failed to read lock file :%v, ", err)
+			return false, err
+		}
+
+		if len(content) <= 0 {
+			log.Errorf("Num bytes read from lock file is 0")
+			return false, fmt.Errorf("Num bytes read from lock file is 0")
+		}
+
+		log.Printf("Read from Lock file:%s", content)
+		// Get the process name if running and
+		// check if that matches with our expected process
+		pName, err := platform.GetProcessNameByID(string(content))
+		if err != nil {
+			return true, nil
```

Comment:
if GetProcessNameByID returned err due to some internal err, we would end up deleting the lock file even if the process which acquired the lock is running.
I think we should return false here. Let's discuss this if you want.